### ISSUE TITLE
Mobile Issues

### ISSUE THREAD

**Issue by bobmorane3000** [michalsnik/aos#261]
hey, i have some issues with disabling AOS @ mobile since it modifies the width of my content and brings my bootstrap-nav-button completely out of position.

so i tried to disable AOS in my custom.js file, but it doesnt work at all - here is the function i used:
	
```
AOS.init({
disable: 'mobile'
});
```
my (bad) solution is to disable all animations with a media querie, but im not happy with that
can anybody explain me what i did wrong? im kinda new to JS.

thank you in advance!

**Comment by jongc** [comment]
That script you have worked for me. Did you enqueue it after `aos.js`?

**Comment by Dzordzu** [comment]
I had same problem. Just try adding this workaround to your code:
```
  var AOSjsFriendly = {
    objects: document.getElementsByClassName("addAos"), //HTML DOM objects with AOS
    disable: 768,
    insertAOS: function() {
      for(var i=0; i<this.objects.length; i++) {
        var obj = this.objects[i];
        var attr = obj.getAttribute("data-addAos");
        obj.setAttribute("data-aos", attr);
      }
    },
    deleteAOS: function() {
      for(var i=0; i<this.objects.length; i++) {
        var obj = this.objects[i];
        obj.removeAttribute("data-aos");
      }
    },
    init: function() {
      if(window.innerWidth>this.disable) this.insertAOS();
    },
    refresh: function() {
      window.addEventListener('resize', debounce(function(){
        if(window.innerWidth>this.disable) AOSjsFriendly.insertAOS();
        else AOSjsFriendly.deleteAOS();
        setTimeout(AOS.refreshHard, 200); //it will refresh AOS position
      }, 20));
    }
  }
```

Then just use: 
```
  AOSjsFriendly.objects = /*Add some DOM with AOS there. By default it will take every DOM with 'addAos' class*/;
  AOSjsFriendly.disable = /*Add breakpoint without unit (default 768px)*/;
  AOSjsFriendly.init(); //this will init AOSjsFriendly
  AOSjsFriendly.refresh(); //this will refresh AOS on every resize event
```

**Comment by 7iomka** [comment]
@Dzordzu, Your example looks great, with one exception. When the page is first loaded, the content will first be displayed to the user, then after adding the class from aos - the styles of the primary hide of the elements will be applied, and only after that the animation will be performed.

**Comment by 7iomka** [comment]
Before this moment I used this tehnique (Example - how to enable AOS only for non-touch devices:
**1 - For styles:**
```
<script defer>

appendStyle("aos.css", notLoadAosForThis());

function notLoadAosForThis(){
  return browserHasTouchEvents();
}
function browserHasTouchEvents(){
  return 'ontouchstart' in window || navigator.maxTouchPoints;
}
function appendStyle(path, falseCondition){
  if(typeof(falseCondition) !== 'undefined' && falseCondition === true) {
    return;
  }

  // my callback function
  // which relies on CSS being loaded
  function CSSDone() {
    console.log('zOMG, CSS is done');
  }
  var url = path,
      head = document.getElementsByTagName('head')[0];
      link = document.createElement('link');

  link.type = "text/css";
  link.rel = "stylesheet"
  link.href = url;

  // MAGIC
  // #1
  link.onload = function () {
    CSSDone('onload listener');
  }
  // #2
  if (link.addEventListener) {
    link.addEventListener('load', function() {
      CSSDone("DOM's load event");
    }, false);
  }
  // #3
  link.onreadystatechange = function() {
    var state = link.readyState;
    if (state === 'loaded' || state === 'complete') {
      link.onreadystatechange = null;
      CSSDone("onreadystatechange");
    }
  };

  // #4
  var cssnum = document.styleSheets.length;
  var ti = setInterval(function() {
    if (document.styleSheets.length > cssnum) {
      // needs more work when you load a bunch of CSS files quickly
      // e.g. loop from cssnum to the new length, looking
      // for the document.styleSheets[n].href === url
      // ...

      // FF changes the length prematurely :()
      CSSDone('listening to styleSheets.length change');
      clearInterval(ti);

    }
  }, 10);

  // MAGIC ends

  head.appendChild(link);
}

  </script>
```

**2 - For script animation rules (not only aos, but initalization of AOS is inside this script)**
```
<script id="site-animations" data-src="basic-animations.min.js" charset="utf-8"></script>
    <script type="text/javascript" defer >
        // for no-touch evenets - add animations
      if(!Modernizr.touchevents){
          console.log('no-touch');
          var animationsScriptNode = document.getElementById('site-animations');
          var srcAttr = animationsScriptNode.getAttribute('data-src');
          animationsScriptNode.setAttribute('src', srcAttr);
        }
    </script>
```

**But I'm still looking for a less verbose option to achieve the same goal.**

**Comment by Dzordzu** [comment]
@7iomka This is why I have added addAos class. You can customize it in every way you will find. So if it's displaying for a user, just change opacity to 0 (or do something similar to that). Please remember about class importing order!
P.S. This approach guarantee best optimization for nonJS websites too!

**Comment by 7iomka** [comment]
@Dzordzu Can you give a demo? If we add a class to elements and set for ex. only opacity to 0, then after adding the data-aos attribute, the type of animation of the element for the first time is knocked down! the simple fade effect works, instead of fade-up for example. 
In a similar way, if we add only a certain transformation to the original class, only one kind of animation will work for the first time.

**Comment by Dzordzu** [comment]
#190 Can possibly have solution to this problem

**Comment by Dzordzu** [comment]
Duplicate of #190 

What I've noticed is fact that possible answer is in given issue

**Comment by Dzordzu** [comment]
Tried, but didn't work. See #190 for the best solution

**Comment by michalsnik** [comment]
You can also:
```
window.addEventListener('load', AOS.refresh)
```
Or just:
```
AOS.init({
  startEvent: 'load',
});
```
